packages/web-components/src/components/va-search-input/va-search-input.tsx

```diff
@@ -0,0 +1,402 @@
+import {
+  Component,
+  Element,
+  Fragment,
+  Host,
+  h,
+  Prop,
+  State,
+  Watch,
+} from '@stencil/core';
+
+@Component({
+  tag: 'va-search-input',
+  styleUrl: 'va-search-input.css',
+  shadow: true,
+})
+export class VaSearchInput {
+  @Element() el: HTMLElement;
+  inputRef!: HTMLInputElement;
+
+  /**
+   * An array of suggestions with suggested text bolded
+   */
+  @State() formattedSuggestions: string[] = [];
+
+  /**
+   * A boolean indicating whether listbox is open or closed
+   */
+  @State() isListboxOpen: boolean;
+
+  /**
+   * Text displayed inside the search button
+   */
+  @Prop() buttonText?: string;
+
+  /**
+   * The aria-label for search input and button. Default is 'Search'.
+   */
+  @Prop() label?: string = 'Search';
+
+  /**
+   * An array of strings containing suggestions to be displayed in listbox.
+   * This component displays up to 5 suggestions.
+   */
+  @Prop() suggestions?: any;
+
+  /**
+   * The value of the input field
+   */
+  @Prop() value?: string = '';
```

Comment:
Having this be `mutable` could make the input handler more intuitive.

Reply:
I believe we could update `value` to be reflective and mutable, but this requires the consumer to handle `value`'s value.

Reply:
When I tried `value` as `reflective` (required to be able to use event.target.value or this value never updates) and `mutable` (this meant updating all value changes to `this.value` instead of `this.inputRef.value`, and also firing events because of the programmatic value changes to the input field), I experienced some weird behavior when NOT managing the state of value in the consumer. If we're okay with having the consumer manage this value so that it's synced, then I can make those changes. I know retrieving the values from non-standard paths like `event.composedPath()[0].value` isn't ideal, but this makes it easier for consumers to use this component without having to manage the state of `value`.

Reply:
Some of the weird behavior included having an initial value of `for`, deleting the text using the Escape key to clear out the input field, and then typing anything would result in the input field being updated to `for[whatever-key-I-typed]`.

Reply:
I've pushed up the changes that would allow consumers to look for `event.target.value` [here](https://github.com/department-of-veterans-affairs/component-library/pull/391), but `value` isn't being reflected properly.

Reply:
This has been addressed [here](https://github.com/department-of-veterans-affairs/component-library/pull/391) although `value` is still not reflected properly. Consumers can retrieve the value of the input in `event.target.value` for both `input` and `submit` events. 

Note: The `value` property will always be correct, but the attribute (if using getAttribute('value')) will be incorrect.